wpilibc/src/main/native/cpp/Tracer.cpp

```diff
@@ -27,7 +41,32 @@ void Tracer::ClearEpochs() {
 
 void Tracer::AddEpoch(std::string_view epochName) {
   auto currentTime = hal::fpga_clock::now();
-  m_epochs[epochName] = currentTime - m_startTime;
+  auto epoch = currentTime - m_startTime;
+  m_epochs[epochName] = epoch;
+  if (m_publishNT) {
+    // Topics are cached with the epoch name as the key, and the publisher
+    // object as the value
+    if (m_publisherCache.count(epochName) == 0) {
+      // Create and prep the epoch publisher
+      auto topic =
+          m_inst.GetIntegerTopic(fmt::format("{}/{}", m_ntTopic, epochName));
+      auto pub = topic.Publish();
+      m_publisherCache.insert(std::pair(epochName, &pub));
+    }
+    m_publisherCache.lookup(epochName)->Set(epoch.count());
```

Comment:
the `IntegerPublisher` returned by `Publish` is stack allocated, and is destroyed when the function exits. The publisher needs to be put on the heap. Also probably should use a smart pointer to allow automatic deallocation.

Reply:
How do place it on the heap or use a smart pointer?